src/hooks/useGovernorName.ts

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from "@tanstack/react-query";
+import Tenant from "@/lib/tenant/tenant";
+
+interface Props {
+  enabled: boolean;
+}
+
+export const GOVERNOR_NAME_QK = "governorName";
+
+export const useGovernorName = ({ enabled }: Props) => {
+  const { contracts } = Tenant.current();
+
+  const { data, isFetching, isFetched } = useQuery({
+    queryKey: [GOVERNOR_NAME_QK, contracts.governor.address],
+    queryFn: async () => {
+      return await contracts.governor.contract.name();
+    },
+    enabled: enabled,
+  });
+
+  return { data, isFetching, isFetched };
+};
```

Comment:
I feel like this is a bit unnecessary. The contract name will never going to change even for upgradable contracts. It might be better to extend the contract config to include the contract name and pull it from there

Reply:
@stepandel In the case of Uniswap and ENS, the contract names differ between testnet and mainnet. This just makes it easier. Also, we don’t have to keep track of those names in the config. My vote is to keep this here.